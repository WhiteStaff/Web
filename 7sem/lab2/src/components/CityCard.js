import React, {Component} from 'react';

export class CityCard extends Component {

    componentWillMount() {
        this.parseJson()
    }

    parseJson() {
        this.coord = this.props.serverInfo.coord;
        this.wind = this.props.serverInfo.wind;
        this.humidity = this.props.serverInfo.humidity;
        this.pressure = this.props.serverInfo.pressure;
        this.clouds = this.props.serverInfo.clouds;
    }


    render() {
        this.parseJson();
            return (
                <div >
                    <div class="alert alert-dark row px-0 mx-1">
                        <div class="col-6">Ветер</div>
                        <div class="col-6 text-right">{this.wind}</div>
                    </div>
                    <div class="alert alert-dark row px-0 mx-1">
                        <div class="col-6">Облачность</div>
                        <div class="col-6 text-right">{this.clouds}</div>
                    </div>
                    <div class="alert alert-dark row px-0 mx-1">
                        <div class="col-6">Давление</div>
                        <div class="col-6 text-right">{this.pressure}</div>
                    </div>
                    <div class="alert alert-dark row px-0 mx-1">
                        <div class="col-6">Влажность</div>
                        <div class="col-6 text-right">{this.humidity}</div>
                    </div>
                    <div class="alert alert-dark row px-0 mx-1">
                        <div class="col-6">Координаты</div>
                        <div class="col-6 text-right">[{this.coord}]</div>
                    </div>
                </div>
            );

    }
}
