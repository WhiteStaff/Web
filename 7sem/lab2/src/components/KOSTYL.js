import React, {Component} from 'react';

export class KOSTYL extends Component {
    componentWillUpdate(nextProps, nextState, nextContext) {
        if(this.props.id){
            this.delete(this.props.id)
        }
    }

    delete(id) {
        // event.preventDefault();
        const data = {
            id: id
        };
        console.log(data);
        this.props.dispatch({
            type: "DELETE_CITY",
            data
        });
    }

}
