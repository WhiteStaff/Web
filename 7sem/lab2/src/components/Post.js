import React, { Component } from 'react';

class Post extends Component {
    render() {
        return (
            <div>
                <h3>{this.props.post.title}</h3>
                <hr />
            </div>
        );
    }
}

export default Post;
