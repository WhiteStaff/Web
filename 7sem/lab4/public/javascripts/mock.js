 var mock = {"coord":{"lon":106.08,"lat":33.77},"weather":[{"id":804,"main":"Clouds","description":"overcast clouds","icon":"04n"}],"base":"model","main":{"temp":275.62,"pressure":1028,"humidity":43,"temp_min":275.62,"temp_max":275.62,"sea_level":1028,"grnd_level":866},"wind":{"speed":0.96,"deg":306},"clouds":{"all":100},"dt":1574080214,"sys":{"country":"CN","sunrise":1574033401,"sunset":1574070701},"timezone":28800,"id":1786288,"name":"Hui","cod":200}
module.exports = mock;
