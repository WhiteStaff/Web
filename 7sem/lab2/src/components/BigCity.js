import React, {Component} from 'react';
import {CityCard} from "./CityCard";

export class BigCity extends Component {
    componentWillMount() {
        this.setState({json: this.props.json});
    }

    render() {
        let name = this.props.json.name;
        let icon = this.props.json.icon;
        let temp = this.props.json.temp;
        let iconLink = "https://openweathermap.org/img/wn/" + icon + "@2x.png";

        if( this.props.json.clouds == 0) return <div> </div>
        else
            return (
                <div class="container pl-0">
                    <div class="row">
                        <div class="col-6">
                            <h2>{name}</h2>
                            <div class="row">
                                <div class="col-3">
                                    <img alt="icon" width="100px" src={iconLink}/>
                                </div>
                                <h1 class="col-6 my-auto display-1 sh">{temp}</h1>
                            </div>
                        </div>
                        <div class="col-6">
                            <CityCard json={this.props.json}/>
                        </div>
                    </div>
                </div>

            );

    }
}
