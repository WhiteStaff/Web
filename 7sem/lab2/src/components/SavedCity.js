import React, {Component} from 'react';
import {CityCard} from "./CityCard";

export class SavedCity extends Component {
    componentWillMount() {
        this.findWeatherDetailsForName(this.props.post.title);
        this.setState({done: false})
    }

    appKey = "2e19bb27bd5e717bac388dc0c1827b17";

    httpRequestAsync(url, callback) {
        console.log("hello");
        var httpRequest = new XMLHttpRequest();
        httpRequest.onreadystatechange = () => {
            if (httpRequest.readyState === 4 && httpRequest.status === 200)
                callback(httpRequest.responseText);
            else if (httpRequest.status === 404) {
                this.setError();
            }
        }
        httpRequest.open("GET", url, true);
        httpRequest.send();
    }
    setError() {
        // alert("Ну ой")
    }

    findWeatherDetailsForName(searchInput) {
        this.setState({done: false});
        if (searchInput === "") {
            // alert("tut")
        } else {
            let searchLink = "https://api.openweathermap.org/data/2.5/weather?q=" + searchInput + "&appid=" + this.appKey;
            // alert(searchInput);
            this.httpRequestAsync(searchLink, (response) => {
                let json = JSON.parse(response);
                this.setState(
                    {
                        json: {
                            name: json.name,
                            icon: json.weather[0].icon,
                            coord: json.coord.lon + ", " + json.coord.lat,
                            wind: json.wind.speed + " m/s",
                            humidity: json.main.humidity + " %",
                            pressure: json.main.pressure + " hpa",
                            clouds: json.weather[0].description
                        },
                        done: true
                    })


            });
        }
    }

    render() {

        if (this.state.done) {
            let name = this.state.json.name;
            let icon = this.state.json.icon;
            let temp = this.state.json.temp;
            let iconLink = "https://openweathermap.org/img/wn/"+ icon +"@2x.png";
            return (
                <div>
                    <h2>{name}</h2>
                    <div>
                        <img alt="icon" src={iconLink}/>
                        <h1>{temp}</h1>
                    </div>
                    <CityCard json={this.state.json}/>
                </div>

            );
        } else {
            return <div> <div>{this.props.post.title}</div>Данные не загружены </div>
        }
    }
}


