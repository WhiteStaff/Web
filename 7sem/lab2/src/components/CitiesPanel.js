import React, {Component} from 'react';
import {connect} from 'react-redux';
import {doAddItem, itemsFetchData, doChangeInput} from "../actions/items";
import SavedCity from "./SavedCity";


class CitiesPanel extends Component {
    handleSubmit = event => {
        event.preventDefault();
        this.props.add(this.props.newCityValue);
    };

    handleChange = (event) => {
        this.props.changeInput(event.target.value);
    };

    render() {
        return (
            <div class="container fav">
                <div>
                    <div class="container pl-0">
                        <div class="row">
                            <div class="col-6 px-0 sh">Избранное</div>
                            <div class="col-6 text-right my-auto form-group">
                                <form onSubmit={this.handleSubmit}>
                                    <input class="favourite-input input_advance" value={this.props.newCityValue}
                                           onChange={this.handleChange} required type="text"
                                           placeholder="Добавить новый город"/>
                                    <button type="buton" class="btn-circle">+</button>
                                </form>
                                <div id="errortext" class="text-center error"></div>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="row">
                    {
                        this.props.items.map((favCity, i = 0) => {
                                i++;
                                return (
                                    [
                                        <SavedCity key={i} json={favCity}/>,
                                        i % 2 === 0 && <div key={-i} className="w-100"/>,
                                    ]
                                )

                            }
                        )
                    }
                </div>
            </div>
        );

    }


}

const mapStateToProps = (state) => {
    return {
        newCityValue: state.newCityValue,
        items: state.items,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        fetchData: (url) => dispatch(itemsFetchData(url)),
        add: (city) => dispatch(doAddItem(city)),
        changeInput: (input) => dispatch(doChangeInput(input))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(CitiesPanel);
