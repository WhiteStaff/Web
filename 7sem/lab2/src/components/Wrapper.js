import React, {Component} from 'react';
import {BigCity} from "./BigCity";
import CitiesPanel from "./CitiesPanel";

export class Wrapper extends Component {

    componentWillMount() {
        this.setState({
            positionAllowed: true, json: {
                coord: "0",
                wind: "0",
                humidity: "0",
                pressure: "0",
                clouds: "0",
                name: "",
                temp: "",
                icon: " "
            },
            done: "noreq",
            inputValue: "",
            error : false
        });
        navigator.geolocation.getCurrentPosition(this.setPosition,this.setFail);


    }

    render() {
        if (this.state.positionAllowed) {
            if (this.state.done) {
                return (
                    <div>
                        <header>
                            <div class="container">
                                <div class="row pl-3">

                                    <div class="col-4">
                                        <span class="sh ">Погода здесь</span>
                                    </div>
                                    <div class="col-3 my-auto mx-auto">
                                        <button class="btn btn-secondary " onClick={() => {
                                            navigator.geolocation.getCurrentPosition(this.setPosition,                                           () => {this.setFail()});
                                        }}>Обновить геолокацию
                                        </button>
                                    </div>

                                </div>
                            </div>
                        </header>
                        <BigCity json={this.state.json} done={this.state.done}/>

                    </div>
                );
            } else {
                return (
                    <div class="container text-center">
                        <div className="row pl-3">
                            <div className="col-4">
                                <span className="sh ">Погода здесь</span>
                            </div>
                            <div className="col-3 my-auto mx-auto">
                                <button className="btn btn-secondary " onClick={() => {
                                    navigator.geolocation.getCurrentPosition((position) => {this.setPosition(position.coords)},
                                        () => {this.setFail()});
                                }}>Обновить геолокацию
                                </button>
                            </div>

                        </div>
                        <h1>Подождите, данные загружаются</h1>
                        <div className="spinner-border m-5" role="status">
                            <span className="sr-only">Loading...</span>
                        </div>

                    </div>

                );
            }
        } else {
            if (this.state.done) {
                return (
                    <div class="container">
                        <div class="row pl-3">
                            <div className="container">
                                <div className="row pl-3">

                                    <div className="col-4">
                                        <span className="sh ">Погода в ДС</span>
                                    </div>
                                    <div className="col-3 my-auto mx-auto">
                                        <button className="btn btn-secondary " onClick={() => {
                                            navigator.geolocation.getCurrentPosition(this.setPosition, () => {
                                                this.setFail()
                                            });
                                        }}>Обновить геолокацию
                                        </button>
                                    </div>

                                </div>
                            </div>
                        </div>
                        <BigCity json={this.state.json}/>
                    </div>
                );

            } else
                return (
                    <div class="container text-center">
                        <div className="row pl-3">
                            <div className="container">
                                <div className="row pl-3">

                                    <div className="col-4">
                                        <span className="sh ">Погода в ДС</span>
                                    </div>
                                    <div className="col-3 my-auto mx-auto">
                                        <button className="btn btn-secondary " onClick={() => {
                                            navigator.geolocation.getCurrentPosition(this.setPosition, () => {
                                                this.setFail()
                                            });
                                        }}>Обновить геолокацию
                                        </button>
                                    </div>

                                </div>
                            </div>

                        </div>
                        <h1>Подождите, данные загружаются</h1>
                        <div className="spinner-border m-5" role="status">
                            <span className="sr-only">Loading...</span>
                        </div>

                    </div>

                );
        }
    }

    getWeather = async (coords) => {
        await this.setState({done: false, error: false});
        console.log(coords);
        fetch('https://api.openweathermap.org/data/2.5/weather?lat=' + coords.latitude + '&lon=' + coords.longitude +
            '&appid=2e19bb27bd5e717bac388dc0c1827b17')
            .then((response) => {
                if (!response.ok) {
                    throw Error(response.statusText);
                }
                return response;

            })
            .then((response) => response.json())
            .then((json) => {
                this.setState(
                    {
                        json: {
                            coord: json.coord.lon + ", " + json.coord.lat,
                            wind: json.wind.speed + " m/s",
                            humidity: json.main.humidity + " %",
                            pressure: json.main.pressure + " hpa",
                            clouds: json.weather[0].description,
                            icon: json.weather[0].icon,
                            name: json.name,
                            temp: parseInt(json.main.temp - 273) + "°C"
                        },
                        done: true,
                        error: false
                    })
            })
            .catch(() => {
                    this.setState(
                        {
                            error: true
                        })
                }
            );

    };

    setPosition = (position) => {
        this.setState({done: false});
        this.getWeather(position.coords);
    };

    coords =
        {
            latitude : 55.75,
            longitude : 37.62
        }

    setFail = () => {
        this.setState({done: false});
        this.getWeather(this.coords);
        this.setState({positionAllowed: false});
    };


}

