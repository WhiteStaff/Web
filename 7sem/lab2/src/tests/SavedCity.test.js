import {shallow} from "../../enzyme";
import {SavedCity} from "../components/SavedCity";
import {shallowToJson} from "enzyme-to-json";
import React from "react";

import mock from './responseExample'
import renderer from "react-test-renderer";

const mocks =
    {
        done: true,
        data : mock.data,
        city : "kyiv"
    };
const mocks2 =
    {
        done: false,
        data : mock.data,
        city : "kyiv"
    };
const mocks3 =
    {
        error: true,
        data : mock.data,
        city : "kyiv"
    };

it('SavedCity renders correctly with load', () => {
    const saved = shallow(<SavedCity key = {0} json={mocks2}/>);
    //saved.setState({done: true, error: false});
    expect(shallowToJson(saved)).toMatchSnapshot();

});
it('SavedCity renders correctly with error', () => {
    const saved = shallow(<SavedCity key = {1} json={mocks3}/>);
    //saved.setState({error: true});
    expect(shallowToJson(saved)).toMatchSnapshot();

});
it('SavedCity renders correctly with data', () => {
    const saved = shallow(<SavedCity key = {1} json={mocks}/>);
    expect(shallowToJson(saved)).toMatchSnapshot();
});
