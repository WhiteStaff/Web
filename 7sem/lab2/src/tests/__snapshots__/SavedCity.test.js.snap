// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedCity renders correctly with data 1`] = `
<div
  class="col-6"
>
  <div
    class="row"
  >
    <div
      class="col-6"
    >
      <h1>
        Shuzenji
      </h1>
    </div>
    <div
      class="col-6 text-right"
    >
      <button
        class="btn-circle"
        onClick={[Function]}
      >
        X
      </button>
    </div>
  </div>
  <h1
    class="my-auto"
  >
    16°C
  </h1>
  <img
    alt="icon"
    src="https://openweathermap.org/img/wn/01n@2x.png"
  />
  <CityCard
    json={
      Object {
        "base": "stations",
        "clouds": Object {
          "all": 2,
        },
        "cod": 200,
        "coord": Object {
          "lat": 35,
          "lon": 139,
        },
        "dt": 1560350192,
        "id": 1851632,
        "main": Object {
          "humidity": 92,
          "pressure": 1009,
          "temp": 289.92,
          "temp_max": 290.93,
          "temp_min": 288.71,
        },
        "name": "Shuzenji",
        "sys": Object {
          "country": "JP",
          "id": 2019346,
          "message": 0.0065,
          "sunrise": 1560281377,
          "sunset": 1560333478,
          "type": 3,
        },
        "timezone": 32400,
        "weather": Array [
          Object {
            "description": "clear sky",
            "icon": "01n",
            "id": 800,
            "main": "Clear",
          },
        ],
        "wind": Object {
          "deg": 107.538,
          "speed": 0.47,
        },
      }
    }
  />
</div>
`;

exports[`SavedCity renders correctly with error 1`] = `
<div
  class="col-6"
>
  <div
    class="row"
  >
    <div
      class="col-6"
    >
      Такого города в известной Вселенной нет
    </div>
    <div
      class="col-6 text-right"
    >
      <button
        class="btn-circle"
        onClick={[Function]}
      >
        X
      </button>
    </div>
  </div>
</div>
`;

exports[`SavedCity renders correctly with load 1`] = `
<div>
  <div>
    Shuzenji
  </div>
  <div>
    Подождите, данные загружаются
  </div>
  <div
    class="spinner-border m-5"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
